code/modules/day_night/day_night_controller.dm

```diff
@@ -0,0 +1,251 @@
+ * is just not fast enough.
+ */
+/datum/day_night_controller/proc/update_area_appearance(light_color, light_alpha)
+	var/mutable_appearance/updated_appearance = mutable_appearance(
+		'icons/effects/daynight_blend.dmi',
+		"white",
+		DAY_NIGHT_LIGHTING_LAYER,
+		LIGHTING_PLANE,
+		light_alpha
+		)
+	updated_appearance.color = light_color
+	area_appearance = updated_appearance
+
+/**
+ * Removes the current mutable appearance from all of the affected areas underlays.
+ */
+/datum/day_night_controller/proc/remove_effect_from_areas()
+	for(var/area/iterating_area as anything in area_cache)
+		iterating_area.underlays -= area_appearance
+
+/**
+ * Applies the current mutable appearance to all of the affected areas underlays.
+ */
+/datum/day_night_controller/proc/apply_effect_to_areas()
+	for(var/area/iterating_area as anything in area_cache)
+		iterating_area.underlays += area_appearance
+
+/**
+ * Intelligently sets the luminosity of our areas providing the light alpha is above the minimum amount of light.
+ * Arguments:
+ * * light_alpha - The new alpha that we are going to be setting.
+ */
+/datum/day_night_controller/proc/set_area_luminosity(light_alpha)
+	var/luminosity = (light_alpha >= MINIMUM_ALPHA_FOR_LUMINOSITY) ? TRUE : FALSE
+	current_luminosity = luminosity
+	for(var/area/iterating_area as anything in area_cache)
+		iterating_area.luminosity = luminosity
+
+/**
+ * Compiles a lookup table using the loaded lightzones for each hour so we can reference it later when switching to said hour.
+ */
+/datum/day_night_controller/proc/compile_transitions()
+	var/hour_index = 0 // We start at 1 as this is 24hr time
+	var/datum/lightzone/current_iterating_lightzone
+	var/transition_value = 0
+	for(var/i in 1 to 24)
+		var/datum/lightzone/check_lightzone = get_lightzone(hour_index)
+		if(current_iterating_lightzone != check_lightzone)
+			current_iterating_lightzone = check_lightzone
+			transition_value = 0
+		var/datum/lightzone/next_lightzone = get_lightzone(current_iterating_lightzone.end_hour)
+		var/segments = ((!current_iterating_lightzone.end_hour ? 24 : current_iterating_lightzone.end_hour) - current_iterating_lightzone.start_hour)
+		var/transition_color = BlendRGB(current_iterating_lightzone.light_color, next_lightzone.light_color, transition_value)
+		var/transition_alpha = (current_iterating_lightzone.light_alpha * (1 - transition_value)) + (next_lightzone.light_alpha * (0 + transition_value))
+		color_lookup_table["[hour_index]"] = transition_color
+		alpha_lookup_table["[hour_index]"] = transition_alpha
+		transition_value += (1 / segments)
+		hour_index++
+
+/**
+ * Gets the corresponding alpha table lookup value
+ * Arguments:
+ * hour - The hour to check
+ * Returns the alpha value
+ */
+/datum/day_night_controller/proc/get_alpha_value(hour)
+	return alpha_lookup_table["[hour]"]
+
+
+// PRESETS
+/datum/day_night_controller/icebox
```

Comment:
shouldn't this default and only preset have a generic name?

Reply:
Well, I wouldn't say so, since the base type is considered abstract, and this is what I choose for the icebox theme.